zebra-chain/src/value_balance.rs

```diff
@@ -0,0 +1,248 @@
+//! A type that can hold the Zcash four types of value pools
+
+use crate::amount::{Amount, Constraint, Error, NegativeAllowed, NonNegative};
+
+/// A structure that hold amounts for the four value pool types.
```

Comment:
Idea: not sure if this makes it clearer, but I thought about rephrasing this to:

```suggestion
/// An amount spread between different Zcash pools.
```